### PR TITLE
(PC-9447) App Native - Sécurité - Appeler la route /user_profiling ap…

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
…rès la validation du téléphone

Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-9447

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/api/gen/.swagger-codegen/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-3.0.27
\ No newline at end of file
+3.0.25
```
Comment (antoinewg) [inline src/api/gen/.swagger-codegen/VERSION:1]: `docker pull swaggerapi/swagger-codegen-cli-v3:latest`

File: src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx [hunk 2]
```diff
@@ -102,6 +107,24 @@ export const SetPhoneValidationCode: FC<SetPhoneValidationCodeProps> = ({ route
...
+    try {
+      Profiling.profileDevice(env.TMX_ORGID, env.TMX_FBSERVER, async (sessionId) => {
+        try {
+          if (!sessionId) {
+            errorMonitoring.captureException(new Error('TMX sessionId is null'))
+            return
+          }
+          await api.postnativev1userProfiling({
+            session_id: sessionId,
+          } as UserProfilingFraudRequest)
```
Comment (antoinewg) [inline src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx:119]: t'as besoin du `as`  ?

File: src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx [hunk 3]
```diff
@@ -102,6 +107,24 @@ export const SetPhoneValidationCode: FC<SetPhoneValidationCodeProps> = ({ route
   )
 
   function onValidateSuccess() {
+    try {
+      Profiling.profileDevice(env.TMX_ORGID, env.TMX_FBSERVER, async (sessionId) => {
```
Comment (antoinewg) [inline src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx:111]: je sais pas si ça change grand chose mais c'est pas un callback async. Donc il faut peut être passer par un `useState`

File: .env.production [hunk 4]
```diff
@@ -85,3 +85,7 @@ APPS_FLYER_DEV_KEY=MzuPoUMwL2khQJHVk8girR
 
 # Id Check
 ID_CHECK_API_URL=https://id-check-middleware.passculture.app
+
+# TMX
+TMX_FBSERVER=server.passculture.app
```
Comment (antoinewg) [inline .env.production:90]:
```suggestion
TMX_FPSERVER=server.passculture.app
```

Comment (antoinewg) [reply .env.production:90]: comme fingerprint

File: src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx (hunk 3 above)
Comment (kopax-polyconseil) [reply src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx:111]: J'ai testé c'est OK, la lib n'attend pas de return.

Comment (kopax-polyconseil) [reply src/features/auth/signup/PhoneValidation/SetPhoneValidationCode.tsx]: J'ai update sans l'async/await

**Review by antoinewg — APPROVED** [review]
Nickel

**Pull request merged**